src/nvim/screen.c

```diff
@@ -1692,6 +1703,16 @@ static void win_update(win_T *wp, DecorProviders *providers)
   wp->w_old_topfill = wp->w_topfill;
   wp->w_old_botfill = wp->w_botfill;
 
+  // Send win_extmarks if needed
+  if (kv_size(win_extmark_arr) > 0) {
+    for (size_t n = 0; n < kv_size(win_extmark_arr); n+=4) {
```

Comment:
if statement is redundant. also `n+=4` should be `n++` I think

Reply:
oops, nice catch!!

Reply:
fixed, and added test for that...